api/src/nv_ingest_api/internal/extract/pdf/engines/nemoretriever.py

```diff
@@ -217,21 +219,30 @@ def nemoretriever_parse_extractor(
     max_workers = nemoretriever_parse_config.workers_per_progress_engine
     with concurrent.futures.ThreadPoolExecutor(max_workers=max_workers) as executor:
 
-        for page_idx in range(page_count):
-            page = doc.get_page(page_idx)
-
-            page_image, padding_offset = _convert_pdfium_page_to_numpy_for_parser(page)
-            pages_for_ocr.append((page_idx, page_image))
-            page_image_for_tables, padding_offset_for_tables = _convert_pdfium_page_to_numpy_for_yolox(page)
-            pages_for_tables.append((page_idx, page_image_for_tables, padding_offset_for_tables))
-
-            page.close()
+        def _render_page(pdf_bytes: bytes, idx: int):
+            with pdfium.PdfDocument(io.BytesIO(pdf_bytes)) as _d:
+                p = _d.get_page(idx)
+                p_img, _ = _convert_pdfium_page_to_numpy_for_parser(p)
+                y_img, y_pad = _convert_pdfium_page_to_numpy_for_yolox(p)
+                p.close()
+            return idx, p_img, y_img, y_pad
+
+        render_workers = min(cpu_count(), 8)
+        render_pool = ThreadPoolExecutor(max_workers=render_workers)
+        render_futs = [
+            render_pool.submit(_render_page, pdf_bytes, i) for i in range(page_count)
+        ]
```

Comment:
@jdye64 ,  I wonder if we can do something like this to first call render pages in parallel, so that our model is not waiting around always ?